simapp/app.go

```diff
@@ -263,11 +263,15 @@ func NewSimApp(
 	memKeys := sdk.NewMemoryStoreKeys(capabilitytypes.MemStoreKey, "testingkey")
 
 	// load state streaming if enabled
-	if _, _, err := streaming.LoadStreamingServices(bApp, appOpts, appCodec, logger, keys); err != nil {
+	streamers, _, err := streaming.LoadStreamingServices(appOpts, appCodec, logger, keys)
+	if err != nil {
 		logger.Error("failed to load state streaming", "err", err)
 		os.Exit(1)
 	}
 
+	// register the streaming service with the BaseApp
+	bApp.SetStreamingService(streamers)
+
```

Comment:
happy to create a helper function to wrap this if deemed necessary

Reply:
Agrees that we should limit the number of basic setter required in app(_v2).go, so if we can limit that this is an improvement.

Reply:
modified, if the change is fine with you want to enable automerging?